test/components/Icon.test.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import { shallow } from 'enzyme';
+
+import Icon from '../../src/Icon';
+
+describe('Icon', () => {
+
+  it('renders correctly', () => {
+    const component = renderer.create(
+      <Icon icon='foo' />
+    );
+    expect(component).toMatchSnapshot();
+  });
+
+  it('renders null if no icon is passed', () => {
+    const component = renderer.create(
+      <Icon />
+    );
+    expect(component.toJSON()).toBeNull();
+  });
+
+  it('renders no child when path is 1', () => {
+    const component = shallow(
+      <Icon icon='foo' paths={1} />
+    );
+    expect(component.children().length).toBe(0);
+  });
+
+  it('renders a child for each path', () => {
+    const component = shallow(
+      <Icon icon='foo' paths={4} />
+    );
+    expect(component.children().length).toBe(4);
+    component.children().forEach((child, i) => {
+      expect(child.prop('className')).toBe(`path${i + 1}`);
+    });
+  });
+
+  it('calls onClick once when clicked', () => {
+    const onClick = jest.fn();
+    const component = shallow(
+      <Icon icon='foo' onClick={onClick} />
+    );
+    component.simulate('click');
+    expect(onClick).toHaveBeenCalledTimes(1);
+  });
+
+  describe('getLocals', () => {
+
+    it('computes className', () => {
+      const icon = new Icon({
+        icon: 'foo',
+        className: 'myicon'
+      });
+      const { className } = icon.getLocals();
+      expect(className).toContain('icon');
+      expect(className).toContain('icon-foo');
+      expect(className).toContain('myicon');
+    });
+
+    it('computes color when not provided', () => {
+      const icon = new Icon({
+        icon: 'foo'
+      });
+      const { style: { color } = {} } = icon.getLocals();
+      expect(color).toBeUndefined();
+    });
+
+    it('computes color when provided as prop', () => {
+      const icon = new Icon({
+        icon: 'foo',
+        color: 'blue'
+      });
+      const { style: { color } = {} } = icon.getLocals();
+      expect(color).toBe('blue');
+    });
+
+    it('computes color when provided as style', () => {
+      const icon = new Icon({
+        icon: 'foo',
+        style: { color: 'blue' }
+      });
+      const { style: { color } = {} } = icon.getLocals();
+      expect(color).toBe('blue');
+    });
+
+    it('overrides color form style if provided as prop', () => {
+      const icon = new Icon({
+        icon: 'foo',
+        style: { color: 'blue' },
+        color: 'red'
+      });
+      const { style: { color } = {} } = icon.getLocals();
+      expect(color).toBe('red');
+    });
+
+  });
+
+});
```

Comment:
yes, probably out-of-scope (even if `atom` automatically adds a new line which means every file has a new line at the end -> we should probably add this rule)